### ISSUE TITLE
programs/cert_write exits with 0 on some errors

### ISSUE THREAD

**Issue by dubek** [ARMmbed/mbedtls#352]
Some errors occurring in cert_write cause it to exit with exitcode 0 (on RHEL 6.7):

```
$ ./cert_write issuer_key=non_existing_file

  . Seeding the random number generator... ok
  . Reading serial number... ok
  . Loading the subject key ... failed
  !  mbedtls_pk_parse_keyfile returned -0x3e00 - PK - Read/write of file failed

$ echo $?
0
```

I'd expect an exitcode which is not 0 to indicate an error occurred. This is important when using these programs in a chain of commands that rely on success of each other (bash `&&` or `set -e` or make).

The reason for this behaviour is that the `main()` function ends with `return(ret)`, but only the least significant byte of `ret` is actually "used" by the OS for the process exitcode. In this case `ret=-0x300` and therefore `ret & 0xff` is 0.

A better solution would be (at the end of `main()`):

```
return(ret == 0 ? 0 : 1);
```

Note that I see this `return(ret)` pattern in other programs as well (for example, `gen_key`, `rsa_verify`, and more).

**Comment by mpg** [comment]
Thanks for your report. I agree with your analysis. Some of our programs already fixed that issue, we're going to generalise the fix. However we happen to be quite loaded now, so please allow some time for that. (Feel free to provide a PR if you'd like to help accelerate the process.)

**Comment by ciarmcom** [comment]
ARM Internal Ref: IOTSSL-548

**Comment by sbutcher-arm** [comment]
This issue has been addressed by PR #532, so can now be closed.
